docs/generate_docs.py

```diff
@@ -0,0 +1,169 @@
+            elif prev_tag == 'color':
+                text += AnsiEscapeCodes.ENDC + ' '
+
+            prev_tag = None
+
+        elif html_element.name in ['kbd']:
+            text += '\n   '
+        # elif html_element.name in ['td']:
+        #     text += '   '  # not sure if its possible to even them out
+        elif html_element.name in ['p']:
+            if 'class' in html_element.attrs:
+                if ' '.join(html_element.attrs['class']) == 'first':
+                    text += ' '
+                else:
+                    text += '\n\n'
+            else:
+                text += '\n\n'
+        elif html_element.name in ['dt']:
+            text += '\n   '
+        elif html_element.name in ['dd']:
+            text += '\n      '
+        elif html_element.name == 'li':
+            if html_element.parent.parent.name == 'li':  # deal with nested lists
+                text += '\n      - '
+            else:
+                text += '\n\n   * '
+        elif html_element.name == 'strong':
+            prev_tag = 'strong'
+            text += '<b>'
+        elif html_element.name == 'pre':
+            if ' '.join(html_element.attrs['class']) == 'code shell literal-block':
+                text += ' '
+            elif ' '.join(html_element.attrs['class']) == 'code literal-block':
+                text += '\n<dim>\n'
+                prev_tag = 'code'
+        elif html_element.name == 'cite':
+            text += ' '
+            prev_tag = 'cite'
+        elif html_element.name == 'span':
+            if ' '.join(html_element.attrs['class']) == 'green':
+                text += ' ' + AnsiEscapeCodes.GREEN
+                prev_tag = 'color'
+            if ' '.join(html_element.attrs['class']) == 'yellow':
+                text += ' ' + AnsiEscapeCodes.YELLOW
+                prev_tag = 'color'
+            if ' '.join(html_element.attrs['class']) == 'red':
+                text += ' ' + AnsiEscapeCodes.RED
+                prev_tag = 'color'
+            if ' '.join(html_element.attrs['class']) == 'grey':
+                text += ' ' + AnsiEscapeCodes.DIM
+                prev_tag = 'color'
+    return text
+
+
+def skip_or_replace_unparseable_directives(rst: str) -> str:
+    rst = rst.replace(':ref:', '')\
+             .replace('.. code-block:: ini', '.. code-block::')
+    return rst
+
+
+def resolve_includes(rst: str, docs_source_path: str) -> str:
+    matches = re.findall(r'\.\. include:: (.*)', rst)
+    for match in matches:
+        with open(f'{docs_source_path}/{match}', 'r') as handle:
+            include_text = handle.read()
+        rst = rst.replace(f'.. include:: {match}', include_text)
+    return rst
+
+
+def skip_prefix_new_lines(txt: str) -> str:
+    txt = re.sub(r'\A[\n]+', '', txt)
+    return txt
+
+
+if __name__ == '__main__':
+    output_docs_path = 'git_machete/long_docs.py'
+    docs_source_path = 'docs/source'
+    output_text = 'from typing import Dict\n\nlong_docs: Dict[str, str] = {\n'
```

Comment:
Actually, while I remember: pls place a comment in the autogenerated file(s) that they're NOT supposed to be edited directly, but instead regenerated via `tox -e docs`